autoscaler/controllers/gateway/hpa.go

```diff
@@ -0,0 +1,75 @@
+package gateway
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/yaml"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	odigosv1 "github.com/odigos-io/odigos/api/odigos/v1alpha1"
+	autoscaling "k8s.io/api/autoscaling/v2beta2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+var (
+	minReplicas = intPtr(1)
+	maxReplicas = int32(10)
+)
+
+func syncHPA(gateway *odigosv1.CollectorsGroup, ctx context.Context, c client.Client, scheme *runtime.Scheme, memConfig *memoryConfigurations) error {
+	logger := log.FromContext(ctx)
+	metricQuantity := resource.MustParse(fmt.Sprintf("%dMi", memConfig.gomemlimitMiB))
+	hpa := &autoscaling.HorizontalPodAutoscaler{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "autoscaling/v2beta2",
+			Kind:       "HorizontalPodAutoscaler",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      gateway.Name,
+			Namespace: gateway.Namespace,
+		},
+		Spec: autoscaling.HorizontalPodAutoscalerSpec{
+			ScaleTargetRef: autoscaling.CrossVersionObjectReference{
+				APIVersion: "apps/v1",
+				Kind:       "Deployment",
+				Name:       gateway.Name,
+			},
+			MinReplicas: minReplicas,
+			MaxReplicas: maxReplicas,
+			Metrics: []autoscaling.MetricSpec{
+				{
+					Type: autoscaling.ResourceMetricSourceType,
+					Resource: &autoscaling.ResourceMetricSource{
+						Name: "memory",
+						Target: autoscaling.MetricTarget{
+							Type:         autoscaling.AverageValueMetricType,
+							AverageValue: &metricQuantity,
+						},
+					},
+				},
+			},
+		},
+	}
+	if err := controllerutil.SetControllerReference(gateway, hpa, scheme); err != nil {
+		logger.Error(err, "Failed to set controller reference")
+		return err
+	}
+
+	hpaBytes, _ := yaml.Marshal(hpa)
+
+	force := true
+	patchOptions := client.PatchOptions{
+		FieldManager: "odigos",
```

Comment:
I am not sure what the semantics for `FieldManager` are for an apply patch. Is it needed?